### PR TITLE
TOC, dev/how-to: add a new tidb-binlog tutorial

### PR DESCRIPTION
**Pull request by huangxiuyan**
Via: https://github.com/pingcap/docs/blob/master/dev/how-to/get-started/tidb-binlog.md
PTAL @lilin90 @GregoryIan

### CODE REVIEW

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+
+该文档假设用户使用的是现代 Linux 发行版本中的 x86-64。示例中使用的是 VMware 中安装运行的 CentOS 7 minimal 版本。强烈建议在一开始就进行清洁安装，以避免受现有环境中未知情况的影响。如果不想使用本地虚拟环境，你也可以使用云服务启动 CentOS 7 VM 版本。
+
+## TiDB-Binlog 介绍
+
+TiDB-Binlog 是一个用于收集 TiDB-Binlog 并提供实时数据备份和同步的工具。TiDB-Binlog 也可用于将 TiDB 集群的数据增量同步到下游。
+
+TiDB-Binlog 支持以下功能场景：
+- 数据增量备份：将 TiDB 集群中的数据复制到另一个集群；或通过 Kafka 发送 TiDB 更新数据并同步到下游。
+- 数据迁移：尤其是将数据从 TiDB 迁移到 MySQL/MariaDB。
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]: Please update the description according to the latest version.

Comment (huangxiuyan) [reply dev/how-to/get-started/TiDB-Binlog-tutorial.md]: Updated.

**Comment by lilin90** [comment]
@CaitinChen PTAL again

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+```
+kolbe@localhost tidb-latest-linux-amd64]$ for p in tidb-server drainer pump tikv-server pd-server; do pkill "$p"; sleep 1; done
+[4]-  Done                    ./bin/tidb-server --config=tidb.toml &>tidb.out
+[5]+  Done                    ./bin/drainer --config=drainer.toml &>drainer.out
+[3]+  Done                    ./bin/pump --config=pump.toml &>pump.out
+[2]+  Done                    ./bin/tikv-server --config=tikv.toml &>tikv.out
+[1]+  Done                    ./bin/pd-server --config=pd.toml &>pd.out
+```
+
+如果需要所有服务退出后重启集群，可以使用一开始启动服务的命令。在上述提及的 [`binlogctl`](#binlogctl) 部分，用户需要先启动 `tidb-server` 再启动 `pump` 之后再启动 `drainer`。
```
Comment (CaitinChen) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
如果需要所有服务退出后重启集群，可以使用一开始启动服务的命令。正如以上 [binlogctl](#binlogctl) 部分所述，需要先启动 Drainer，再启动 Pump，最后启动 `tidb-server`。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+./bin/pd-server --config=pd.toml &>pd.out &
+./bin/tikv-server --config=tikv.toml &>tikv.out &
+./bin/drainer --config=drainer.toml &>drainer.out &
+sleep 3
+./bin/pump --config=pump.toml &>pump.out &
+sleep 3
+./bin/tidb-server --config=tidb.toml &>tidb.out &
+```
+
+如果有组件启动失败尝试单独重启该组件。
```
Comment (CaitinChen) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
如果有组件启动失败，请尝试单独重启该组件。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+
+## 总结
+
+该文档介绍了如何通过设置 TiDB Binlog 、使用单个 Pump 和 Drainer 组成的集群同步 TiDB 集群的数据到下游的 MariaDB 服务器。可以发现，TiDB Binlog 是用于获取处理 TiDB 集群中更新数据的综合性平台工具。
```
Comment (CaitinChen) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
本文档介绍了如何通过设置 TiDB Binlog、使用单个 Pump 和 Drainer 组成的集群同步 TiDB 集群的数据到下游的 MariaDB 服务器。可以发现，TiDB Binlog 是用于获取处理 TiDB 集群中更新数据的综合性平台工具。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+
+在更稳健的开发、测试或生产部署环境中，可以使用多个 TiDB 服务器以实现高可用性和扩展性。使用多个 Pump 实例可以确保发送到 TiDB 实例的应用流量不受 Pump 集群中问题的影响。或者可以使用另加的 Drainer 实例同步数据到不同的下游以及实现数据增量备份。
```
Comment (CaitinChen) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
在更稳健的开发、测试或生产部署环境中，可以使用多个 TiDB 服务器以实现高可用性和扩展性。使用多个 Pump 实例可以确保发送到 TiDB 实例的应用流量不受 Pump 集群中问题的影响。或者可以使用另加的 Drainer 实例发送数据到不同的下游或实现数据增量备份。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 6]
```diff
@@ -0,0 +1,414 @@
...
+
+TiDB Binlog 集群由 **Pump** 和 **Drainer** 两个组件组成。一个 Pump 集群中有若干个 Pump 节点。每个 Pump 节点连接到 TiDB 实例，接收 TiDB 集群中每个 TiDB 实例的更新数据。Drainer 连接到 Pump 集群，并将接收到的更新数据转换到某个特定下游（例如 Kafka、另一个 TiDB 集群或 MySQL 或 MariaDB 服务器）指定的正确格式。 
+
+![TiDB Binlog architecture](/media/tidb_binlog_cluster_architecture.png)
+
+Pump 的集群架构确保 TiDB 或 Pump 集群中有新的实例加入或退出时更新数据不会丢失。
+
+## 安装
+
+由于 RHEL/CentOS 7 的默认包装库中包括 MariaDB server，本示例选择的是 MariaDB Server。之后的使用中，除了服务器，也需要客户端。安装指令如下：
```
Comment (CaitinChen) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
由于 RHEL/CentOS 7 的默认包装库中包括 MariaDB Server，本示例选择的是 MariaDB Server。之后的使用中，除了服务器，也需要客户端。安装指令如下：
```

File: TOC.md [hunk 7]
```diff
@@ -14,6 +14,7 @@
       - [使用 Docker Compose](dev/how-to/get-started/local-cluster/install-from-docker-compose.md)
     - [SQL 基本操作](dev/how-to/get-started/explore-sql.md)
     - [读取历史数据](dev/how-to/get-started/read-historical-data.md)
+    - [TiDB Binlog 部署](dev/how-to/get-started/TiDB-Binlog-tutorial.md)
```
Comment (lilin90) [inline TOC.md]:
```suggestion
    - [TiDB Binlog 部署](dev/how-to/get-started/tidb-binlog.md)
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 8]
```diff
@@ -0,0 +1,419 @@
+---
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]: Please rename the `TiDB-Binlog-tutorial.md` to `tidb-binlog.md`. Generally, we'll keep the name in docs-cn with that in docs.

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 9]
```diff
@@ -0,0 +1,419 @@
...
+title: 部署 TiDB Binlog 
+category: how-to
+---
+
+# 部署 TiDB Binlog
+
+本文档将先介绍，在将数据迁移到 MariaDB 实例之前，如何在 Placement Driver、TiKV、TiDB、Pump 或 Drainer 各个组件中的单个节点上部署 TiDB Binlog。
+
+希望上手实践 TiDB Binlog 工具的用户需要对 [TiDB 架构](https://pingcap.com/docs/architecture/)有一定的了解，最好有创建过 TiDB 集群的经验。该文档也有助于你简单快速了解 TiDB Binlog 架构以及相关概念。
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
希望上手实践 TiDB Binlog 工具的用户需要对 [TiDB 架构](/architecture.md)有一定的了解，最好有创建过 TiDB 集群的经验。该文档也有助于你简单快速了解 TiDB Binlog 架构以及相关概念。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 10]
```diff
@@ -0,0 +1,419 @@
...
+
+TiDB Binlog 是一个用于收集 TiDB 中二进制日志数据提供实时数据备份和同步的工具。TiDB Binlog 也可用于将 TiDB 集群的数据增量同步到下游。
+
+TiDB Binlog 支持以下功能场景：
+
+- 数据增量备份：将 TiDB 集群中的数据同步到另一个集群；或通过 Kafka 发送 TiDB 更新数据并同步到下游。
+- 数据迁移：将数据从 MySQL 或者 MariaDB 迁移到 TiDB 上。在这种情况下你可以使用 TiDB Data Migration (DM) 从 MySQL 或 MariaDB 集群中获取数据，并同步到 TiDB。之后可用 TiDB Binlog 让独立的下游 MySQL 或 MariaDB 实例或集群与 TiDB 集群保持同步。
+- 数据同步：将发送到 TiDB 的应用流量同步更新到下游的 MySQL 或 MariaDB 实例或集群。即使流量数据迁移到 TiDB 过程中出现问题，在 MySQL 或 MariaDB 中也能撤回该流量数据，且不会造成宕机或数据损失。
+
+更多信息参考 [TiDB Binlog Cluster 版本用户文档](https://pingcap.com/docs-cn/v2.1/tools/TiDB-Binlog-cluster/)。
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
更多信息参考 [TiDB Binlog Cluster 版本用户文档](https://pingcap.com/docs-cn/tools/tidb-binlog-cluster/)。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 11]
```diff
@@ -0,0 +1,419 @@
...
+
+
+1. 启动 `drainer` 
+
+    ```bash
+    sudo systemctl start mariadb
+    ./bin/drainer --config=drainer.toml &>drainer.out &
+    ```
+
+    你可以在更简单的操作系统中安装 MySQL，只需要保证通过 3306 端口监听，且可作为 “root” 用户及空密码连接到 MySQL，或必要时可调整 drainer.toml。
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
    你可以在更简单的操作系统中安装 MySQL，只需要保证通过 3306 端口监听，且可作为 "root" 用户及空密码连接到 MySQL，或必要时可调整 `drainer.toml`。
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 12]
```diff
@@ -0,0 +1,419 @@
...
+
+    ```sql
+    create database tidbtest;
+    use tidbtest;
+    create table t1 (id int unsigned not null auto_increment primary   key);
+    insert into t1 () values (),(),(),(),();
+    select * from t1;
+    ```
+
+    预期输出：
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
    预期输出：
    
```

File: dev/how-to/get-started/TiDB-Binlog-tutorial.md [hunk 13]
```diff
@@ -0,0 +1,419 @@
...
+    |  5 |
+    +----+
+    5 rows in set (0.00 sec)
+    ```
+
+    可以看到查询 MariaDB 时插入到 TiDB 相同的行数据。TiDB Binlog 部署成功完成。
+
+## binlogctl
+
+加入到集群的 Pump 和 Drainer 的数据存储在 Placement Driver (PD) 中。binlogctl 可用于查询和修改状态信息。更多信息请参考 [binlogctl guide](https://pingcap.com/docs-cn/tools/binlog/operation/)。
```
Comment (lilin90) [inline dev/how-to/get-started/TiDB-Binlog-tutorial.md]:
```suggestion
加入到集群的 Pump 和 Drainer 的数据存储在 Placement Driver (PD) 中。binlogctl 可用于查询和修改状态信息。更多信息请参考 [binlogctl guide](https://pingcap.com/docs-cn/tools/binlog/operation/#binlogctl-工具)。 
```

**Comment by huangxiuyan** [comment]
@lilin90 @CaitinChen PTAL, thanks!

File: dev/how-to/get-started/tidb-binlog.md [hunk 14]
```diff
@@ -0,0 +1,420 @@
...
+    endpoints=["127.0.0.1:2379"]
+    [rocksdb]
+    max-open-files=1024
+    [raftdb]
+    max-open-files=1024
+    ```
+
+## 启动程序
+
+现在可启动各个组件。推荐启动顺序依次为 Placement Driver （PD）、TiKV、Pump（TiDB 发送 binlog 日志必须连接 Pump 服务）、TiDB。
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
现在可启动各个组件。推荐启动顺序依次为 Placement Driver (PD)、TiKV、Pump（TiDB 发送 binlog 日志必须连接 Pump 服务）、TiDB。
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 15]
```diff
@@ -0,0 +1,420 @@
...
+    [kolbe@localhost tidb-latest-linux-amd64]$ jobs
+    [1]   Running                 ./bin/pd-server --config=pd.toml &>pd.out &
+    [2]   Running                 ./bin/tikv-server --config=tikv.toml &>tikv.out &
+    [3]-  Running                 ./bin/pump --config=pump.toml &>pump.out &
+    [4]+  Running                 ./bin/tidb-server --config=tidb.toml &>tidb.out &
+    ```
+
+    如果有服务启动失败（例如出现 “`Exit 1`” 而不是 “`Running`”），尝试重启单个服务。 
+
+    按以上步骤操作后，TiDB 的 4 个 组件应该都已开始运行。
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
    按以上步骤操作后，TiDB 的 4 个组件应该都已开始运行。
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 16]
```diff
@@ -0,0 +1,420 @@
...
+
+
+1. 启动 `drainer` 
+
+    ```bash
+    sudo systemctl start mariadb
+    ./bin/drainer --config=drainer.toml &>drainer.out &
+    ```
+
+    你可以在更简单的操作系统中安装 MySQL，只需要保证通过 3306 端口监听，且可作为 "root" 用户及空密码连接到 MySQL，或必要时可调整 `drainer.toml`。
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
    你可以在更简单的操作系统中安装 MySQL，只需要保证通过 3306 端口监听，且可作为使用空密码的 "root" 用户连接到 MySQL，或必要时可调整 `drainer.toml`。
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 17]
```diff
@@ -0,0 +1,420 @@
...
+
+3. 启动 PD 之后（但在启动 Drainer 和 Pump 之前）使用 binlogctl 更新已暂定 Drainer 的状态
+
+    ```
+    ./bin/binlogctl --pd-urls=http://127.0.0.1:2379 --cmd=update-drainer --node-id=localhost.localdomain:8249 --state=offline
+    ```
+
+## 清理
+
+在 shell 终端里可启动创建集群的所有进程（pd-server 、tikv-server、pump、tidb-server、drainer）。可通过在 shell 终端中执行 `pkill -P $$` 停止 TiDB 集群服务和 TiDB Binlog 进程。按一定的顺序停止这些进程有利于留出足够的时间彻底关闭每个组件。
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
在 shell 终端里可启动创建集群的所有进程（`pd-server`、`tikv-server`、`pump`、`tidb-server`、`drainer`）。可通过在 shell 终端中执行 `pkill -P $$` 停止 TiDB 集群服务和 TiDB Binlog 进程。按一定的顺序停止这些进程有利于留出足够的时间彻底关闭每个组件。
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 18]
```diff
@@ -0,0 +1,420 @@
...
+```
+kolbe@localhost tidb-latest-linux-amd64]$ for p in tidb-server drainer pump tikv-server pd-server; do pkill "$p"; sleep 1; done
+[4]-  Done                    ./bin/tidb-server --config=tidb.toml &>tidb.out
+[5]+  Done                    ./bin/drainer --config=drainer.toml &>drainer.out
+[3]+  Done                    ./bin/pump --config=pump.toml &>pump.out
+[2]+  Done                    ./bin/tikv-server --config=tikv.toml &>tikv.out
+[1]+  Done                    ./bin/pd-server --config=pd.toml &>pd.out
+```
+
+如果需要所有服务退出后重启集群，可以使用一开始启动服务的命令。正如以上 [`binlogctl`](#binlogctl) 部分所述，需要先启动 Drainer 再启动 Pump, 最后启动 `tidb-server`。
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
如果需要所有服务退出后重启集群，可以使用一开始启动服务的命令。正如以上 [`binlogctl`](#binlogctl) 部分所述，需要先启动 Drainer 再启动 Pump，最后启动 `tidb-server`。
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 19]
```diff
@@ -0,0 +1,420 @@
...
+100  368M  100  368M    0     0  8394k      0  0:00:44  0:00:44 --:--:-- 11.1M
+[kolbe@localhost ~]$ cd tidb-latest-linux-amd64
+[kolbe@localhost tidb-latest-linux-amd64]$
+```
+
+## 配置
+
+执行以下步骤配置一个 TiDB 集群，该集群包括 `pd-server`、`tikv-server` 和 `tidb-server` 各组件的单个实例。
+
+1. 填充配置文件
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]: Please add a period or comma here.

File: dev/how-to/get-started/tidb-binlog.md [hunk 20]
```diff
@@ -0,0 +1,420 @@
...
+
+本文档将先介绍，在将数据迁移到 MariaDB 实例之前，如何在 Placement Driver、TiKV、TiDB、Pump 或 Drainer 各个组件中的单个节点上部署 TiDB Binlog。
+
+希望上手实践 TiDB Binlog 工具的用户需要对 [TiDB 架构](/architecture.md)有一定的了解，最好有创建过 TiDB 集群的经验。该文档也有助于你简单快速了解 TiDB Binlog 架构以及相关概念。
+
+> **警告：**
+>
+> 该文档中部署 TiDB 的操作指导**不适用于**在生产或研发环境中部署 TiDB 的情况。
+
+该文档假设用户使用的是现代 Linux 发行版本中的 x86-64。示例中使用的是 VMware 中运行的 CentOS 7 最小化安装。建议在一开始就进行清洁安装，以避免受现有环境中未知情况的影响。如果不想使用本地虚拟环境，你也可以使用云服务启动 CentOS 7 VM。
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
该文档假设用户使用的是现代 Linux 发行版本中的 x86-64。示例中使用的是 VMware 中运行的 CentOS 7 最小化安装。建议在一开始就进行清洁安装，以避免受现有环境中未知情况的影响。如果不想使用本地虚拟环境，也可以使用云服务启动 CentOS 7 VM。
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 21]
```diff
@@ -0,0 +1,420 @@
+---
+title: 部署 TiDB Binlog
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
title: TiDB Binlog 教程
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 22]
```diff
@@ -0,0 +1,420 @@
+---
+title: 部署 TiDB Binlog 
+category: how-to
+---
+
+# 部署 TiDB Binlog
```
Comment (CaitinChen) [inline dev/how-to/get-started/tidb-binlog.md]:
```suggestion
# TiDB Binlog 教程
```

File: dev/how-to/get-started/tidb-binlog.md [hunk 23]
```diff
@@ -0,0 +1,419 @@
...
+该文档假设用户使用的是现代 Linux 发行版本中的 x86-64。示例中使用的是 VMware 中运行的 CentOS 7 最小化安装。建议在一开始就进行清洁安装，以避免受现有环境中未知情况的影响。如果不想使用本地虚拟环境，也可以使用云服务启动 CentOS 7 VM。
+
+## TiDB Binlog 简介
+
+TiDB Binlog 用于收集 TiDB 中二进制日志数据、提供实时数据备份和同步以及将 TiDB 集群的数据增量同步到下游。
+
+TiDB Binlog 支持以下功能场景：
+
+- 数据增量备份：将 TiDB 集群中的数据增量同步到另一个集群；或通过 Kafka 发送 TiDB 更新数据并同步到下游。
+- 数据迁移：将数据从 MySQL 或者 MariaDB 迁移到 TiDB 上。在这种情况下可以使用 TiDB Data Migration (DM) 从 MySQL 或 MariaDB 集群中获取数据，并同步到 TiDB。之后可用 TiDB Binlog 让独立的下游 MySQL 或 MariaDB 实例或集群与 TiDB 集群保持同步。
```
Comment (WangXiangUSTC) [inline dev/how-to/get-started/tidb-binlog.md]: 这里应该不用提把 mysql 的数据同步到 tidb 和 DM

File: dev/how-to/get-started/tidb-binlog.md [hunk 24]
```diff
@@ -0,0 +1,419 @@
...
+- 数据同步：将发送到 TiDB 的应用流量同步更新到下游的 MySQL 或 MariaDB 实例或集群。即使流量数据迁移到 TiDB 过程中出现问题，在 MySQL 或 MariaDB 中也能撤回该流量数据，且不会造成宕机或数据损失。
```
Comment (WangXiangUSTC) [inline dev/how-to/get-started/tidb-binlog.md]: 没太看懂数据同步和上面数据迁移的区别

File: dev/how-to/get-started/tidb-binlog.md [hunk 25]
```diff
@@ -0,0 +1,419 @@
...
+
+更多信息参考 [TiDB Binlog Cluster 版本用户文档](https://pingcap.com/docs-cn/tools/tidb-binlog-cluster/)。
+
+## 架构
+
+TiDB Binlog 集群由 **Pump** 和 **Drainer** 两个组件组成。一个 Pump 集群中有若干个 Pump 节点。每个 Pump 节点连接到 TiDB 实例，接收 TiDB 集群中每个 TiDB 实例的更新数据。Drainer 连接到 Pump 集群，并将接收到的更新数据转换到某个特定下游（例如 Kafka、另一个 TiDB 集群或 MySQL 或 MariaDB Server）指定的正确格式。
```
Comment (WangXiangUSTC) [inline dev/how-to/get-started/tidb-binlog.md]: 是 TiDB 实例主动连接的 Pump，tidb 主动把 binlog 数据发送到 pump

File: dev/how-to/get-started/tidb-binlog.md [hunk 26]
```diff
@@ -0,0 +1,419 @@
...
+
+本文档中的 binlogctl 主要用于集群重启。如果在 TiDB 集群中终止并尝试重启所有的进程，由于 Pump 无法连接 Drainer 且认为 Drainer 依旧“在线”，Pump 会拒绝启动。这里的进程并不包括下游的 MySQL 或 MariaDB 或 Drainer。
+
+以下有三个方案可解决上述问题：
+
+1. 使用 binlogctl 停止 Drainer，而不是结束进程
+
+    ```
+    ./bin/binlogctl --pd-urls=http://127.0.0.1:2379 --cmd=drainers
+    ./bin/binlogctl --pd-urls=http://127.0.0.1:2379 --cmd=offline-drainer --node-id=localhost.localdomain:8249
```
Comment (WangXiangUSTC) [inline dev/how-to/get-started/tidb-binlog.md]: 例子里用 offline-drainer 不太好，建议改成 pause-drainer

File: dev/how-to/get-started/tidb-binlog.md [hunk 27]
```diff
@@ -0,0 +1,419 @@
...
+    ```
+
+2. 在启动 Pump **之前**先启动 Drainer
+
+3. 在启动 PD 之后但在启动 Drainer 和 Pump 之前，使用 binlogctl 更新已暂定 Drainer 的状态
+
+    ```
+    ./bin/binlogctl --pd-urls=http://127.0.0.1:2379 --cmd=update-drainer --node-id=localhost.localdomain:8249 --state=offline
```
Comment (WangXiangUSTC) [inline dev/how-to/get-started/tidb-binlog.md]: 建议把 offline 改成 paused

**Comment by huangxiuyan** [comment]
@WangXiangUSTC @lilin90 @CaitinChen PTAL, updated mostly the overview based on what I leared about binlog these days, in order to help Chinese readers understand clearly. If it is not clear enough, I'll quickly confirm again with Morgan.

**Comment by WangXiangUSTC** [comment]
LGTM

File: dev/how-to/get-started/tidb-binlog.md (hunk 24 above)
Comment (WangXiangUSTC) [reply dev/how-to/get-started/tidb-binlog.md]: 还是感觉这个地方提 DM 不太好，不过原作者坚持保留的话就这样吧

Comment (huangxiuyan) [reply dev/how-to/get-started/tidb-binlog.md]: kolbe 想保留的是从 MySQL/MariaDB 迁移到 TiDB，但他前面提到了 “TiDB Binlog is particularly useful”，而我们用的是 DM从 MySQL/MariaDB 迁移到 TiDB，所以这里就有理解冲突的问题。读者会想有了 DM 迁移为什么还要用 TiDB Binlog？或许应用场景比较丰富，这里又扩展开来，应该是涉及到上游迁移和下游同步的两个过程，涉及到两个工具~如果后续有更好的版本我们还可以再次修改！

**Pull request merged**